models/helpers.js

```diff
@@ -33,13 +33,9 @@ passwordSchema
 
 module.exports = { omit, pick, hash, passwordSchema };
 
-
 Object.assign(mongoose.Model, {
-  getAllById(arrayOfIds) {
-    return this.find({ _id: { $in: arrayOfIds } });
-  },
-
-  getPublicById(arrayOfIds) {
-    return this.find({ _id: { $in: arrayOfIds }, type: Types.Public });
+  getById(arrayOfIds, type) {
+    const conditions = { _id: { $in: arrayOfIds }, type };
+    return this.find(omitBy(conditions, prop => !prop));
```

Comment:
Create generic method:
```js
findMany(query = {}) { /*...*/ }
```
And handle query properties in following way:

- If property `value` is array use `prop: { $in: value }` form, otherwise use simply `prop: value`
- If property `value` is '*' remove it from `query` (using omitBy)

Omitting stuff by checking for _falsey_ value works in this case but in general it is bad idea because asking for `undefined` and/or `false` value inside query is legit.